app/controllers/nodes_controller.rb

```diff
@@ -18,6 +18,42 @@ def show
     @minion = Minion.find(params[:id])
   end
 
+  # rubocop:disable Metrics/AbcSize
+  def assign
+    @minion = Minion.find(params[:node_id])
+
+    unless Minion.roles.keys.include?(params[:role])
+      raise Minion::InvalidRole, "Could not assign role #{params[:role]}"
+    end
+
+    respond_to do |format|
+      if @minion.assign_role(params[:role])
+        res, body = Velum::Salt.orchestrate
+        http_status = res.code == "200" ? :ok : :unprocessable_entity
+        flash[:error] = "Salt failed to orchestrate: #{body.inspect}" unless http_status == :ok
+
+        format.html { redirect_to nodes_path }
+        format.json { head http_status }
+      else
+        format.html do
+          flash[:error] = @minion.errors.full_messages.first
+          redirect_to nodes_path
+        end
+        format.json { render json: @minion.errors, status: :unprocessable_entity }
+      end
+    end
+  rescue Velum::SaltApi::SaltConnectionException,
+         ActiveRecord::RecordNotFound,
+         Minion::InvalidRole => e
```

Comment:
makes sense, but then i think we'd have to write another rendering block, which i wanted to avoid. and it would look almost the same as the one in the rescue section i believe. this raise and rescue is also an early return. is this somehow bad practice?